src/components/ui/TextareaEditor/TextareaEditor.js

```diff
@@ -106,77 +63,33 @@ class TextareaEditor extends PureComponent<Props, State> {
   }
 
   state = {
-    id: this.props.id || 'default',
-    text: this.props.text || '',
-    html: this.props.html || '',
-    shouldShowTextLength: false,
-    options: {
-      placeholder: {
-        text: this.props.placeholder || ''
-      },
-      disableEditing: this.props.disableEditing || false,
-      toolbar: !this.props.disableEditing
-    }
-  }
-
-  changeShouldShowTextLength = (val: boolean) => {
-    this.setState(() => ({ shouldShowTextLength: val }))
+    value: this.props.value || ''
   }
 
-  handleChange = (text: string, html: string) => {
-    this.setState(() => ({ text, html }))
-    const { onChange } = this.props
-    if (typeof onChange === 'function') {
-      onChange(text, html)
-    }
-  }
-
-  handleFocus = (target: HTMLTextAreaElement) => {
-    const { text, html } = this.state
-    const { onFocus } = this.props
-
-    this.changeShouldShowTextLength(true)
-
-    if (typeof onFocus === 'function') {
-      onFocus(target, text, html)
-    }
-  }
-
-  handleBlur = (target: HTMLTextAreaElement) => {
-    const { text, html } = this.state
-    const { onBlur } = this.props
-
-    this.changeShouldShowTextLength(false)
-
-    if (typeof onBlur === 'function') {
-      onBlur(target, text, html)
+  handleChangeValue = (input: Object): void => {
+    const inputValue = input.target.value
+    if (inputValue.length <= this.props.charLimit) {
+      this.setState({ value: inputValue })
     }
   }
 
   render () {
-    const { charLimit, linesLimit } = this.props
-    const { id, text, html, shouldShowTextLength, options } = this.state
-
-    const counterVisibilityClassName = shouldShowTextLength ? '' : 'hidden'
-    const editableClassName = options.disableEditing ? '' : editableClass
+    const { id, name, placeholder, charLimit, linesLimit } = this.props
+    const { value } = this.state
 
     return Root(
       <div>
-        <div className={`${editorContainerStyles} ${editableClassName}`}>
-          <MediumEditorWrapper
-            text={text}
-            html={html}
-            id={id}
-            charLimit={charLimit}
-            linesLimit={linesLimit}
-            options={options}
-            onChange={this.handleChange}
-            onFocus={this.handleFocus}
-            onBlur={this.handleBlur}
-          />
-        </div>
+        <InputField
+          id={id}
+          name={name}
+          type='textarea'
+          placeholder={placeholder}
+          value={value}
+          linesLimit={linesLimit}
+          onChange={this.handleChangeValue}
+        />
```

Comment:
You can use:

```js
const { charLimit, ...rest } = this.props
```
and then declare this way:
```suggestion
        <InputField
          {...rest}        
          type='textarea'
          onChange={this.handleChangeValue}
        />
```

That would eliminate the redundancy on the type settings and props definitions of this component.